date-adder.js

```diff
@@ -0,0 +1,23 @@
+const add = (date, diff) => {
+  inc = Number(diff.split(/(\D)/)[0])
+  type = (diff.split(/(\D)/)[1])
+  if (type === 's') {
+    return new Date(date.setSeconds(date.getSeconds() + inc));
```

Comment:
all the setters update the original date. maybe try

```js
const d = new Date(date)
if(type === 's') return d.setSeconds(d.getSeconds() + inc))
```

that way you only update the new date and not the original date